src/app/tabs/compile-tab.js

```diff
@@ -217,6 +238,31 @@ class CompileTab extends ViewPlugin {
     }
   }
 
+  /**
+   * set the compiler configuration
+   * This function is used by remix-plugin compiler API.
+   * @param {object} settings {evmVersion, optimize, compilerUrl, version, language}
+   */
+  setCompilerConfig (settings) {
+    return new Promise((resolve, reject) => {
+      addTooltip(yo`<div><b>${this.currentRequest.from}</b> is updating the <b>Solidity compiler configuration</b>.<pre class="text-left">${JSON.stringify(settings, null, '\t')}</pre></div>`)
+      this.compilerContainer.setLanguage(settings.language)
+      this.compilerContainer.setEvmVersion(settings.evmVersion)
+      this.compilerContainer.setOptimize(settings.optimize)
+      this.compilerContainer.setVersion(settings.version)
```

Comment:
sould maybe compilerContainer has setSettings?

Reply:
yes sure, the thing is that these methods were already existing before